tests/table/table_io/test_read_time_cast.py

```diff
@@ -28,9 +28,9 @@
         ),
         # Test reordering of columns
         (
-            pa.Table.from_pydict({"foo": pa.array([1, 2, 3]), "bar": pa.array([1, 2, 3])}),
+            pa.Table.from_pydict({"foo": pa.array([1, 2, 3]), "bar": pa.array([4, 5, 6])}),
             Schema._from_field_name_and_types([("bar", DataType.int64()), ("foo", DataType.int64())]),
-            MicroPartition.from_pydict({"bar": pa.array([1, 2, 3]), "foo": pa.array([1, 2, 3])}),
+            MicroPartition.from_pydict({"bar": pa.array([4, 5, 6]), "foo": pa.array([1, 2, 3])}),
```

Comment:
Why?

Reply:
To make sure that we catch the case where the names are flipped but not the data haha